components/calculation/MeasureHighlightingPanel.tsx

```diff
@@ -12,20 +18,146 @@ const useStyles = createStyles({
   panel: {
     maxHeight: '100%',
     overflow: 'scroll'
+  },
+  highlightRed: {
+    backgroundColor: '#edd8d0',
+    color: '#a63b12',
+    borderBottomColor: '#a63b12',
+    borderBottomStyle: 'double'
+  },
+  highlightGreen: {
+    backgroundColor: '#ccebe0',
+    color: '#20744c',
+    borderBottomColor: '#20744c',
+    borderBottomStyle: 'solid'
   }
 });
 
 export interface MeasureHighlightingPanelProps {
   patientId: string;
 }
 
+export interface PopulationResult {
+  [key: string]: string | number;
+}
+
+export interface DesiredPopulations {
+  [key: string]: number;
+}
+export interface BothPopulations {
+  [key: string]: {
+    desired: number;
+    actual: number | undefined;
+  };
+}
+
 export default function MeasureHighlightingPanel({ patientId }: MeasureHighlightingPanelProps) {
   const { classes } = useStyles();
+  const measureBundle = useRecoilValue(measureBundleState);
   const measureReportLookup = useRecoilValue(measureReportLookupState);
+  const currentPatients = useRecoilValue(patientTestCaseState);
+  const [opened, setOpened] = useState(false);
+  const measure = useMemo(() => {
+    return measureBundle.content?.entry?.find(e => e.resource?.resourceType === 'Measure')?.resource as fhir4.Measure;
+  }, [measureBundle]);
 
-  return (
-    <>
-      <div className={classes.highlightedMarkup}>{parse(measureReportLookup[patientId]?.text?.div || '')}</div>
-    </>
-  );
+  /**
+   * Creates an object where the population is the key and the value is 0 or 1:
+   * 0 if that population is not a desired population and 1 if it is.
+   */
+  function constructDesiredPopulationsValuesArray(measurePopulations: MultiSelectData[]) {
+    const desiredPopulations: DesiredPopulations = {};
+    measurePopulations.map(measurePopulation => {
+      if (currentPatients[patientId].desiredPopulations?.includes(measurePopulation.value)) {
+        desiredPopulations[measurePopulation.label as string] = 1;
+      } else {
+        desiredPopulations[measurePopulation.label as string] = 0;
+      }
+    });
+
+    return desiredPopulations;
+  }
+
+  /**
+   * Creates an object where the population is the key and it has two properties: actual and desired.
+   * Both properties can either be 0 or 1: 0 if that population is not desired (or actual) and 1 if
+   * that population is desired (or actual).
+   */
+  function constructBothPopulationsValuesArray(group: MeasureReportGroup | undefined) {
+    const desiredPopulations = constructDesiredPopulationsValuesArray(getMeasurePopulations(measure));
+    const bothPopulations: BothPopulations = {};
+    group?.population?.map(population => {
+      const key = population?.code?.coding?.[0]?.display || population?.code?.coding?.[0]?.code;
+      if (key) {
+        bothPopulations[key as string] = { desired: desiredPopulations[key as string], actual: population?.count };
+      }
+    });
```

Comment:
A little unconventional to have a `.map` here with no return value. I would swap this to a forEach, or if you're feeling really fancy, go for a reduce. This comment also applies to the above function